### DISCUSSION TITLE
Changing a flex item to "bock"

### DISCUSSION THREAD

**Discussion by sergio101** [tailwindlabs/tailwindcss#4426] [Help] [answered]
<img width="870" alt="image" src="https://user-images.githubusercontent.com/77505/119199316-d9450680-ba58-11eb-803d-a040e6e1f249.png">

I am brand new to tailwindcss, and i am trying to make a responsive form. I am using flex and enclosing the whole form in a flex. When the form goes to `sm` i would like to make a few flex elements block-level elements, so I can "stack" the block-level elements. I tried adding "block" to the title element, but am having no luck. any idea how I can do this? this is the last hurdle before launch.. thanks!

This is how it looks as I go smaller:

<img width="708" alt="image" src="https://user-images.githubusercontent.com/77505/119199589-5a040280-ba59-11eb-8ca4-99e27de0b830.png">

**Comment by olets** [comment]
The basic idea is to nest flexboxes. Start by grouping what you want to have on the same line on mobile, and then group those into the lines you want on larger screens.

Here's a very minimal example. When it's narrow it's two rows, when it's wide it's one. https://play.tailwindcss.com/k8NcMNy8Vm

**Comment by dance2die** [comment, marked as answer]
If you started with mobile-first design, then you could start with `flex flex-col` (for `xs/sm`) and only when `md` and up is hit, you can set the `md:flex-row`.

```html
<main class="py-6 border border-b-4 flex 
    flex-col justify-center items-center gap-4 
    md:flex-row md:max-gap-6">
...
</main>

```
**demo**: https://play.tailwindcss.com/sqETCa0EN9?size=722x843

![flex demo](https://user-images.githubusercontent.com/8465237/119244842-762f9e80-bb42-11eb-98fe-d7aa3a9aa0b6.gif)

**Reply by simonswiss** [reply]
This is a great answer! The way to "stack" flex children elements is to change the `flex-direction` on the flex wrapper from `row` to `column` with `flex-col`.

Adding `display: block` to a flex child will have no effect since it's still going to behave according to its `flex` parent.

Thanks for that demo, @dance2die ✨

**Reply by simonswiss** [reply]
And [as @olets mentioned](https://github.com/tailwindlabs/tailwindcss/discussions/4426#discussioncomment-772317), if you need a more complex stacking with sometimes more than one element per "stack", nesting flex containers is the way to go 👍

**Reply by olets** [reply]
👍 just depends on exactly what the OP means by "make a few flex elements block-level elements"
